src/main/java/pl/touk/sputnik/Main.java

```diff
@@ -31,8 +36,17 @@ public static void main(String[] args) {
         }
 
         ConfigurationHolder.initFromFile(commandLine.getOptionValue(CliOption.CONF.getCommandLineParam()));
-        ConfigurationHolder.instance().updateWithCliOptions(commandLine);
-        ConnectorFacade facade = ConnectorFacadeFactory.INSTANCE.build(ConfigurationHolder.instance().getProperty(GeneralOption.CONNECTOR_TYPE));
+        Configuration configuration = ConfigurationHolder.instance();
+        configuration.updateWithCliOptions(commandLine);
+
+        ConnectorType connectorType = ConnectorType.valueOf(configuration.getProperty(GeneralOption.CONNECTOR_TYPE));
```

Comment:
Changed the extractor so it is NPE-safe and when unsupported connector is passed, default one is returned. Now it works like other methods.